src/App.tsx

```diff
@@ -1,99 +1,30 @@
-import { useState } from "react";
-import TransferButtons from "./TransferButtons";
-import List from "./List";
-import {
-  intersection,
-  not,
-} from "./utils";
-import { ListInput } from "./components/list-input/ListInput";
+import React, {useState} from "react";
 import './App.css';
+import {Tabs} from "./components/tabs/Tabs";
+import TransferListApp from "./apps/transfer-list/TransferListApp";
+import {DragAndDropList} from "./components/drag-and-drop-list/DragAndDropList";
+import {DragAndDropApp} from "./apps/drag-and-drop/DragAndDropApp";
+import {TicTacToeApp} from "./apps/tic-tac-toe/TicTacToeApp";
 
-export type ListItem =  { 
-  id: string; 
-  name: string; 
+
+const tabComponents: Record<number, React.FC> = {
+  0: TransferListApp,
+  1: DragAndDropApp,
+  2: TicTacToeApp,
 };
 
-const initialListItems: Array<ListItem> = [
-  { id: "1", name: "item 1" },
-  { id: "2", name: "item 2" },
-]
 
 function App() {
-  const [firstListItems, setFirstListItems] = useState<Array<ListItem>>(initialListItems);
-  const [secondListItems, setSecondListItems] = useState<Array<ListItem>>([]);
-  const [checkedItems, setChecktedItems] = useState<Array<ListItem>>([]);
-  
-  const leftCheckedItems = intersection(checkedItems, firstListItems);
-  const rightCheckedItems = intersection(checkedItems, secondListItems);
-
-  function handleCheck(item: ListItem, checked: boolean) {
-    if(checked){
-      setChecktedItems((prev) => [...prev, item]);
-    }
-    else {
-      setChecktedItems((prev) => prev.filter((prevItem) => prevItem.id !== item.id));
-    }
-  }
-
-  function handleLeftMove() {
-    setFirstListItems(firstListItems.concat(rightCheckedItems));
-    setSecondListItems(not(secondListItems, rightCheckedItems));
-    setChecktedItems(not(checkedItems, rightCheckedItems));
-  }
-
-  function handleRightMove() {
-    setSecondListItems(secondListItems.concat(leftCheckedItems));
-    setFirstListItems(not(firstListItems, leftCheckedItems));
-    setChecktedItems(not(checkedItems, leftCheckedItems));
-  }
-
-  function handleAddToList(item: string) {
-    setFirstListItems((prev) => [...prev, { name: item, id: crypto.randomUUID()}]);
-  }
+  const [activeIdx, setActiveIdx] = useState<number>(0)
+  const ActiveComponent = tabComponents[activeIdx];
```

Comment:
Instead of doing it here like this, I would install react router dom and assign each app to a different route.

Reply:
yeah it we want to assign seperate urls for each app that might be useful. 👍